content/posts/2021-06-25-Treasury-Expenditure-Bug.md

```diff
@@ -0,0 +1,167 @@
+
+Due to the test treasury spend transaction mined in mainnet on [block 556,716](https://explorer.dcrdata.org/tx/7507bcc72bfde895065034e12e6d462f2360163cd0c879f0db35514f9456b2c1), the maximum amount that can now be spent from the decentralized treasury account for the next few months is approximately 0.15 DCR, which is too low to be able to fulfill all monthly invoices in the CMS system.
+
+While no development team enjoys finding issues in shipped software and doubly so in distributed consensus code, we need to highlight that Decred's governance model and the way the decentralized treasury functions were introduced worked as designed to prevent a bigger problem.
+
+  - The migration plan from the legacy treasury to the decentralized treasury was created to handle any cases such as this.
+  - The payments to contractors will still continue as usual, coming from the legacy treasury wallet.
+  - There are no risks of the network forking, given the corresponding on-chain vote already activated the new rules and all older versions of the software are not following the main chain anymore.
+  - The decentralized treasury account is still accruing its 10% block reward.
+  - Finally, all the funds in the legacy treasury wallet are still intact and available for spending and eventually being moved to the new account system.
+
+
+The rest of this document will provide more technical details about the issue, how it was introduced, how it will be fixed and its impact to the network.
+
+# In-Depth Analysis
+
+For the rest of this document we assume the reader has some basic knowledge about how the new decentralized treasury feature works and how changes to the consensus rules of Decred are proposed, coded, deployed and activated.
+
+## The History of the Expenditure Policy in the Code
+
+The initial Politeia proposal for the [decentralized treasury](https://proposals-archive.decred.org/proposals/c96290a) included a provision to limit the total amount of funds that could be withdrawn from the treasury during a period of time. To quote from the proposal:
+
+> `OP_TSPEND` shall [...] prevent spending more than this months treasury accrued value + 50%
+
+This was further refined in some comments and will be generally referred to as the "maximum expenditure policy check".
+
+The motivation for adding this restriction is to avoid overspending the treasury - that is, to avoid withdrawing too much DCR from the treasury account in a short period of time.
+
+This has two main applications:
+
+  - A security and safety mechanism, so that in case of implementation bugs or compromise of treasury keys the treasury account wouldn't be depleted too quickly, before the issue could be addressed.
+  - A generic "fiscal" policy over the treasury's funds, so that it would not be amenable to being overly spent by short term interests.
```

Comment:
```suggestion
- A generic "fiscal" policy over the treasury's funds, so that it would not be amenable to being overly spent by short-term interests.
```